ext/openssl/ossl.c

```diff
@@ -1150,9 +1150,29 @@ Init_openssl(void)
     /*
      * Version number of OpenSSL the ruby OpenSSL extension was built with
      * (base 16)
+     *
+     * OpenSSL 3: 0xMNN00PP0 (major minor 00 patch 0)
+     * OpenSSL 1: 0xMNNFFPPS (major minor fix patch status)
+     * LibreSSL : 0x20000000 (fixed value)
+     *
+     * See also the man page OPENSSL_VERSION_NUMBER(3).
      */
     rb_define_const(mOSSL, "OPENSSL_VERSION_NUMBER", INT2NUM(OPENSSL_VERSION_NUMBER));
 
+    /*
+     * Version number of LibreSSL the ruby OpenSSL extension was built with
+     * (base 16)
+     *
+     * LibreSSL: 0xMNNFFPPS (major minor fix patch status)
+     *
+     * See also the man page OPENSSL_VERSION_NUMBER(3).
+     */
+#if defined(LIBRESSL_VERSION_NUMBER)
+    rb_define_const(mOSSL, "LIBRESSL_VERSION_NUMBER", INT2NUM(LIBRESSL_VERSION_NUMBER));
+#else
+    rb_define_const(mOSSL, "LIBRESSL_VERSION_NUMBER", Qnil);
+#endif
```

Comment:
Should we just not define `OpenSSL::LIBRESSL_VERSION_NUMBER` when it's not compiled with LibreSSL?

Reply:
Okay. I would agree with your opinion. I was not familiar with how the LibreSSL specific features are implemented. Seeing the following code below, there are methods defined in a specific conditionally. So, I fix fix this part not to define `OpenSSL::LIBRESSL_VERSION_NUMBER` in the case the Ruby OpenSSL is not compiled with LibreSSL.

https://github.com/ruby/openssl/blob/db633c52f7b364ded8185b661fb557e9db2f6383/ext/openssl/ossl.c#L1227-L1256

Reply:
I rebased this PR. Now we only defined the `OpenSSL::LIBRESSL_VERSION_NUMBER` in LibreSSL case.